tiledb/sm/array_schema/array_schema.cc

```diff
@@ -1023,7 +1023,14 @@ void ArraySchema::add_dimension_label(
 
     // Create the dimension label reference.
     auto dim_label_ref = make_shared<DimensionLabel>(
-        HERE(), dim_id, name, uri, dim, label_order, label_type);
+        HERE(),
+        dim_id,
+        name,
+        uri,
+        dim,
+        label_order,
+        label_type,
+        domain_->memory_tracker());
```

Comment:
Should we use `ArraySchema::memory_tracker_` here instead? If so it might be good to merge @abigalekim's PR for ArraySchema first and I can update this after. This is the only usage of `Domain::memory_tracker()` so I could also remove that accessor.